pkg/apis/apps/types.go

```diff
@@ -187,6 +187,12 @@ type StatefulSetStatus struct {
 	// updateRevision, if not empty, indicates the version of the StatefulSet used to generate Pods in the sequence
 	// [replicas-updatedReplicas,replicas)
 	UpdateRevision string
+
+	// collisionCount is the count of hash collisions for the StatefulSet. The StatefulSet controller
+	// uses this field as a collision avoidance mechanism when it needs to create the name for the
+	// newest ControllerRevision.
+	// +optional
+	CollisionCount *int64
```

Comment:
Personally I don't think int64 is necessary, but it's the type for the field in other controller objects, e.g., https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/extensions/types.go#L345 and https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/extensions/types.go#L519. We want the field to have consistent type across controllers.